File: Weatley.Backend/Weatley.Model/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Weatley.Model.Entities
{
    public class Order : IEntityBase
    {
        public Guid Id { get; set; }
        public double FinalPrice { get; set; }
        public string Comment { get; set; }
        public DateTime OrderDate { get; set; }
        public Customer Customer { get; set; }
        public string Status { get; set; }
        public string StatusComment { get; set; }
        public string signalRId { get; set; }

        //Relation with Product
        public ICollection<ProductOrdered> ProductsOrdered { get; set; }
    }
}

File: Weatley.App/app/core/entities/order.ts
import { Product } from './product';
import { Customer } from './customer';
import { ProductOrdered } from './product-ordered';

export class Order {
	id: string;
	finalPrice: number;
	comment: string;
	orderDate: Date;
	customer: Customer;
	productsOrdered: ProductOrdered[];
	status: string;
	statusComment: string;
	signalRId: string;
	countProducts: number;
	totalProducts: string;

	constructor(values: Object = {}) {
		Object.assign(this, values);
	}
}
